providers/providers.go

```diff
@@ -482,7 +523,11 @@ func InstallIO(reader io.ReadCloser, conf InstallConf) ([]*Provider, error) {
 	// so we don't spam the system with unnecessary data. Optionally we could
 	// keep them and re-use them, so they don't have to download again.
 	defer func() {
-		if err = os.RemoveAll(tmpdir); err != nil {
+		// We don't set a max retry, since we can indefinitely try to remove this
+		err := osRetry(func() error {
```

Comment:
could that end up in an infinite loop?

Reply:
yes. If you feel better, I could give it a high limit?

Reply:
an upper bound would be great.

Reply:
added an upper bound. I don't think it's helpful in this particular case, but it's definitely something we can learn as we go. These boundaries are estimates based on the data that is processed, curious to see how we tweak it over time.